src/components/ColorInput/index.js

```diff
@@ -0,0 +1,372 @@
+        disabled,
+        error,
+        tabIndex,
+        name,
+        placeholder,
+        bottomHelpText,
+        onFocus,
+        onBlur: blurInProps,
+    } = useReduxForm(props);
+    const [isOpen, setIsOpen] = useState(false);
+    const [colorValue, setColorValue] = useState(getColorValue(value));
+    const [sampleColor, setSampleColor] = useState(value);
+    const pickerRef = useRef();
+    const containerRef = useRef();
+    const triggerRef = useRef();
+    const alphaInputRef = useRef();
+    const inputRef = useRef();
+
+    const inputId = useUniqueIdentifier('color-input');
+    const errorMessageId = useErrorMessageId(error);
+    const labelId = useLabelId(label);
+
+    const [focusIndex, setFocusIndex] = useFocusIndex(
+        containerRef,
+        triggerRef,
+        inputRef,
+        alphaInputRef,
+    );
+
+    useOutsideClick(
+        containerRef,
+        () => {
+            setFocusIndex(-1);
+        },
+        focusIndex > -1,
+    );
+    useOutsideClick(
+        pickerRef,
+        event => {
+            if (
+                event.target !== triggerRef.current &&
+                !triggerRef.current.contains(event.target) &&
+                !pickerRef.current.contains(event.target)
+            ) {
+                setIsOpen(false);
+            }
+        },
+        isOpen,
+    );
+    useWindowResize(() => setIsOpen(false), isOpen);
+
+    const onBlur = event => {
+        const { hex } = value;
+        if (!isHexColor(hex)) setSampleColor(undefined);
+        blurInProps(event);
+    };
+    const handleFocus = useHandleFocus(focusIndex, onFocus, setFocusIndex, value);
+    const handleBlur = useHandleBlur(focusIndex, onBlur, value);
+
+    const handleClick = event => {
+        if (focusIndex === 1) {
+            setFocusIndex(0);
+        } else {
+            setFocusIndex(1);
+        }
+
+        event.preventDefault();
+        if (isOpen) setIsOpen(false);
+        else setIsOpen(true);
+        return onClick(event);
+    };
+
+    const handleChange = event => {
+        const eventValue = event.target.value;
+        const { alpha } = value;
+        const hex = getHexString(eventValue);
+        const isValid = isHexColor(hex);
+        const newValue = { hex, alpha, isValid };
+        if (isValid) {
+            setSampleColor(newValue);
+            setColorValue(getColorValue(newValue));
+        }
+        onChange(newValue);
+    };
+
+    const handleAlphaChange = event => {
+        if (!event.target.value) {
+            onChange({ ...value, alpha: null });
+            return;
+        }
+
+        let alpha = Number.parseInt(event.target.value || '0', 10);
+        if (alpha > 100) alpha = 100;
+        else if (alpha < 0) alpha = 0;
+        alpha /= 100;
+
+        const newValue = { ...value, alpha };
+        setSampleColor(newValue);
+        setColorValue(getColorValue(newValue));
+        if (!Number.isNaN(alpha)) onChange(newValue);
+    };
```

Comment:
more test case to check this:
1. should send 100 when type an alpha greater than 100
2. should send 0 when type an alpha less than 0